iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/Networks.java

```diff
@@ -241,59 +236,64 @@ private static void addDisonnectedShunt(Terminal.BusBreakerView view, double q,
-        table.addCell(Integer.toString(balanceMainCC.disconnectedLoads.size()));
-        table.addCell(Integer.toString(balanceOtherCC.connectedLoads.size()));
-        table.addCell(Integer.toString(balanceOtherCC.disconnectedLoads.size()));
-        table.addCell("Load (MW)");
-        table.addCell(Double.toString(balanceMainCC.connectedLoadVolume));
-        table.addCell(Double.toString(balanceMainCC.disconnectedLoadVolume));
-        table.addCell(Double.toString(balanceOtherCC.connectedLoadVolume));
-        table.addCell(Double.toString(balanceOtherCC.disconnectedLoadVolume));
-        table.addCell("Generator count");
-        table.addCell(Integer.toString(balanceMainCC.connectedGenerators.size()));
-        table.addCell(Integer.toString(balanceMainCC.disconnectedGenerators.size()));
-        table.addCell(Integer.toString(balanceOtherCC.connectedGenerators.size()));
-        table.addCell(Integer.toString(balanceOtherCC.disconnectedGenerators.size()));
-        table.addCell("Max generation (MW)");
-        table.addCell(Double.toString(balanceMainCC.connectedMaxGeneration));
-        table.addCell(Double.toString(balanceMainCC.disconnectedMaxGeneration));
-        table.addCell(Double.toString(balanceOtherCC.connectedMaxGeneration));
-        table.addCell(Double.toString(balanceOtherCC.disconnectedMaxGeneration));
-        table.addCell("Generation (MW)");
-        table.addCell(Double.toString(balanceMainCC.connectedGeneration));
-        table.addCell(Double.toString(balanceMainCC.disconnectedGeneration));
-        table.addCell(Double.toString(balanceOtherCC.connectedGeneration));
-        table.addCell(Double.toString(balanceOtherCC.disconnectedGeneration));
-        table.addCell("Shunt at nom V (MVar)");
-        table.addCell(Double.toString(balanceMainCC.connectedShuntPositiveVolume) + " " +
-                Double.toString(balanceMainCC.connectedShuntNegativeVolume) +
-                " (" + Integer.toString(balanceMainCC.connectedShunts.size()) + ")");
-        table.addCell(Double.toString(balanceMainCC.disconnectedShuntPositiveVolume) + " " +
-                Double.toString(balanceMainCC.disconnectedShuntNegativeVolume) +
-                " (" + Integer.toString(balanceMainCC.disconnectedShunts.size()) + ")");
-        table.addCell(Double.toString(balanceOtherCC.connectedShuntPositiveVolume) + " " +
-                Double.toString(balanceOtherCC.connectedShuntNegativeVolume) +
-                " (" + Integer.toString(balanceOtherCC.connectedShunts.size()) + ")");
-        table.addCell(Double.toString(balanceOtherCC.disconnectedShuntPositiveVolume) + " " +
-                Double.toString(balanceOtherCC.disconnectedShuntNegativeVolume) +
-                " (" + Integer.toString(balanceOtherCC.disconnectedShunts.size()) + ")");
-        return table;
+    private static Writer writeInTable(ConnectedPower balanceMainCC, ConnectedPower balanceOtherCC) {
```

Comment:
Return a ˋString` instead of a Writer: writer.toString()

Reply:
Alright :+1: